src/shogun/mathematics/Statistics.cpp

```diff
@@ -342,13 +342,13 @@ SGVector<int32_t> Statistics::sample_indices(int32_t sample_size, int32_t N, PRN
 	{
 		rnd=uniform_int_dist(prng, {1, i});
 		if (rnd<sample_size)
-			permuted_idxs[rnd]=idxs[i];
+			result[rnd]=idxs[i];
```

Comment:
again this will definitely come up as a compile error if you would have compiled it locally....

Reply:
I am compling it as we talk cause it takes alot of time locally but yea I will change that sorry

Reply:
yeah but then wait till it finishes before you push.
do you have `ccache` installed locally?

Reply:
@vigsterkr yes I do